File: RegexTester/My/MyProject.cs
using System.Windows.Forms;

namespace RegexTester.My
{
    using Microsoft.VisualBasic;
    using Microsoft.VisualBasic.ApplicationServices;
    using Microsoft.VisualBasic.CompilerServices;
    using RegexTester;
    using System;
    using System.CodeDom.Compiler;
    using System.Collections;
    using System.ComponentModel;
    using System.ComponentModel.Design;
    using System.Diagnostics;
    using System.Reflection;
    using System.Runtime.CompilerServices;
    using System.Runtime.InteropServices;

    [StandardModule, GeneratedCode("MyTemplate", "8.0.0.0"), HideModuleName]
    internal sealed class MyProject
    {
        private static readonly ThreadSafeObjectProvider<MyComputer> m_ComputerObjectProvider = new ThreadSafeObjectProvider<MyComputer>();
        private static readonly ThreadSafeObjectProvider<MyApplication> m_AppObjectProvider = new ThreadSafeObjectProvider<MyApplication>();
        private static readonly ThreadSafeObjectProvider<Microsoft.VisualBasic.ApplicationServices.User> m_UserObjectProvider = new ThreadSafeObjectProvider<Microsoft.VisualBasic.ApplicationServices.User>();
        private static ThreadSafeObjectProvider<MyForms> m_MyFormsObjectProvider = new ThreadSafeObjectProvider<MyForms>();
        private static readonly ThreadSafeObjectProvider<MyWebServices> m_MyWebServicesObjectProvider = new ThreadSafeObjectProvider<MyWebServices>();

        [HelpKeyword("My.Computer")]
        internal static MyComputer Computer
        {
            [DebuggerHidden]
            get
            {
                return m_ComputerObjectProvider.GetInstance;
            }
        }

        [HelpKeyword("My.Application")]
        internal static MyApplication Application
        {
            [DebuggerHidden]
            get
            {
                return m_AppObjectProvider.GetInstance;
            }
        }

        [HelpKeyword("My.User")]
        internal static Microsoft.VisualBasic.ApplicationServices.User User
        {
            [DebuggerHidden]
            get
            {
                return m_UserObjectProvider.GetInstance;
            }
        }

        [HelpKeyword("My.Forms")]
        internal static MyForms Forms
        {
            [DebuggerHidden]
            get
            {
                return m_MyFormsObjectProvider.GetInstance;
            }
        }

        [HelpKeyword("My.WebServices")]
        internal static MyWebServices WebServices
        {
            [DebuggerHidden]
            get
            {
                return m_MyWebServicesObjectProvider.GetInstance;
            }
        }

        [EditorBrowsable(EditorBrowsableState.Never), MyGroupCollection("System.Windows.Forms.Form", "Create__Instance__", "Dispose__Instance__", "My.MyProject.Forms")]
        internal sealed class MyForms
        {
            public RegexTester.AboutBoxForm m_AboutBoxForm;
            public RegexTester.CompileForm m_CompileForm;
            public RegexTester.EscapeForm m_EscapeForm;
            public RegexTester.GenerateCodeForm m_GenerateCodeForm;
            public RegexTester.MainForm m_MainForm;
            public RegexTester.PropertiesForm m_PropertiesForm;
            [ThreadStatic]
            private static Hashtable m_FormBeingCreated;

            [DebuggerHidden]
            private static T Create__Instance__<T>(T Instance) where T: Form, new()
            {
                if ((Instance == null) || Instance.IsDisposed)
                {
                    if (m_FormBeingCreated != null)
                    {
                        if (m_FormBeingCreated.ContainsKey(typeof(T)))
                        {
                            throw new InvalidOperationException(Utils.GetResourceString("WinForms_RecursiveFormCreate", new string[0]));
                        }
                    }
                    else
                    {
                        m_FormBeingCreated = new Hashtable();
                    }
                    m_FormBeingCreated.Add(typeof(T), null);
                    try
                    {
                        return Activator.CreateInstance<T>();
                    }
                    catch 
                    {
                    }
                    finally
                    {
                        m_FormBeingCreated.Remove(typeof(T));
                    }
                }
                return Instance;
            }

            [DebuggerHidden]
            private void Dispose__Instance__<T>(ref T instance) where T: Form
            {
                instance.Dispose();
                instance = default(T);
            }

            [EditorBrowsable(EditorBrowsableState.Never)]
            public override bool Equals(object o)
            {
                return base.Equals(RuntimeHelpers.GetObjectValue(o));
            }

            [EditorBrowsable(EditorBrowsableState.Never)]
            public override int GetHashCode()
            {
                return base.GetHashCode();
            }

            [EditorBrowsable(EditorBrowsableState.Never)]
            internal Type GetType()
            {
                return typeof(MyProject.MyForms);
            }

            [EditorBrowsable(EditorBrowsableState.Never)]
            public override string ToString()
            {
                return base.ToString();
            }

            public RegexTester.AboutBoxForm AboutBoxForm
            {
                get
                {
                    this.m_AboutBoxForm = Create__Instance__<RegexTester.AboutBoxForm>(this.m_AboutBoxForm);
                    return this.m_AboutBoxForm;
                }
                set
                {
                    if (value != this.m_AboutBoxForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.AboutBoxForm>(ref this.m_AboutBoxForm);
                    }
                }
            }

            public RegexTester.CompileForm CompileForm
            {
                get
                {
                    this.m_CompileForm = Create__Instance__<RegexTester.CompileForm>(this.m_CompileForm);
                    return this.m_CompileForm;
                }
                set
                {
                    if (value != this.m_CompileForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.CompileForm>(ref this.m_CompileForm);
                    }
                }
            }

            public RegexTester.EscapeForm EscapeForm
            {
                get
                {
                    this.m_EscapeForm = Create__Instance__<RegexTester.EscapeForm>(this.m_EscapeForm);
                    return this.m_EscapeForm;
                }
                set
                {
                    if (value != this.m_EscapeForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.EscapeForm>(ref this.m_EscapeForm);
                    }
                }
            }

            public RegexTester.GenerateCodeForm GenerateCodeForm
            {
                get
                {
                    this.m_GenerateCodeForm = Create__Instance__<RegexTester.GenerateCodeForm>(this.m_GenerateCodeForm);
                    return this.m_GenerateCodeForm;
                }
                set
                {
                    if (value != this.m_GenerateCodeForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.GenerateCodeForm>(ref this.m_GenerateCodeForm);
                    }
                }
            }

            public RegexTester.MainForm MainForm
            {
                get
                {
                    this.m_MainForm = Create__Instance__<RegexTester.MainForm>(this.m_MainForm);
                    return this.m_MainForm;
                }
                set
                {
                    if (value != this.m_MainForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.MainForm>(ref this.m_MainForm);
                    }
                }
            }

            public RegexTester.PropertiesForm PropertiesForm
            {
                get
                {
                    this.m_PropertiesForm = Create__Instance__<RegexTester.PropertiesForm>(this.m_PropertiesForm);
                    return this.m_PropertiesForm;
                }
                set
                {
                    if (value != this.m_PropertiesForm)
                    {
                        if (value != null)
                        {
                            throw new ArgumentException("Property can only be set to Nothing");
                        }
                        this.Dispose__Instance__<RegexTester.PropertiesForm>(ref this.m_PropertiesForm);
                    }
                }
            }
        }

        [EditorBrowsable(EditorBrowsableState.Never), MyGroupCollection("System.Web.Services.Protocols.SoapHttpClientProtocol", "Create__Instance__", "Dispose__Instance__", "")]
        internal sealed class MyWebServices
        {
            [DebuggerHidden]
            private static T Create__Instance__<T>(T instance) where T: new()
            {
                if (instance == null)
                {
                    return Activator.CreateInstance<T>();
                }
                return instance;
            }

            [DebuggerHidden]
            private void Dispose__Instance__<T>(ref T instance)
            {
                instance = default(T);
            }

            [DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)]
            public override bool Equals(object o)
            {
                return base.Equals(RuntimeHelpers.GetObjectValue(o));
            }

            [EditorBrowsable(EditorBrowsableState.Never), DebuggerHidden]
            public override int GetHashCode()
            {
                return base.GetHashCode();
            }

            [DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)]
            internal Type GetType()
            {
                return typeof(MyProject.MyWebServices);
            }

            [DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)]
            public override string ToString()
            {
                return base.ToString();
            }
        }

        [ComVisible(false), EditorBrowsable(EditorBrowsableState.Never)]
        internal sealed class ThreadSafeObjectProvider<T> where T: new()
        {
            [ThreadStatic, CompilerGenerated]
            private static T m_ThreadStaticValue;

            internal T GetInstance
            {
                [DebuggerHidden]
                get
                {
                    if (MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue == null)
                    {
                        MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue = Activator.CreateInstance<T>();
                    }
                    return MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue;
                }
            }
        }
    }
}

File: RegexTester/My/MyProject.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.ApplicationServices
Imports Microsoft.VisualBasic.CompilerServices
Imports RegexTester
Imports System
Imports System.CodeDom.Compiler
Imports System.Collections
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Diagnostics
Imports System.Reflection
Imports System.Runtime.CompilerServices
Imports System.Runtime.InteropServices

Namespace RegexTester.My
    <StandardModule, GeneratedCode("MyTemplate", "8.0.0.0"), HideModuleName> _
    Friend NotInheritable Class MyProject
        ' Properties
        <HelpKeyword("My.Computer")> _
        Friend Shared ReadOnly Property Computer As MyComputer
            <DebuggerHidden> _
            Get
                Return MyProject.m_ComputerObjectProvider.GetInstance
            End Get
        End Property

        <HelpKeyword("My.Application")> _
        Friend Shared ReadOnly Property Application As MyApplication
            <DebuggerHidden> _
            Get
                Return MyProject.m_AppObjectProvider.GetInstance
            End Get
        End Property

        <HelpKeyword("My.User")> _
        Friend Shared ReadOnly Property User As User
            <DebuggerHidden> _
            Get
                Return MyProject.m_UserObjectProvider.GetInstance
            End Get
        End Property

        <HelpKeyword("My.Forms")> _
        Friend Shared ReadOnly Property Forms As MyForms
            <DebuggerHidden> _
            Get
                Return MyProject.m_MyFormsObjectProvider.GetInstance
            End Get
        End Property

        <HelpKeyword("My.WebServices")> _
        Friend Shared ReadOnly Property WebServices As MyWebServices
            <DebuggerHidden> _
            Get
                Return MyProject.m_MyWebServicesObjectProvider.GetInstance
            End Get
        End Property


        ' Fields
        Private Shared ReadOnly m_ComputerObjectProvider As ThreadSafeObjectProvider(Of MyComputer) = New ThreadSafeObjectProvider(Of MyComputer)
        Private Shared ReadOnly m_AppObjectProvider As ThreadSafeObjectProvider(Of MyApplication) = New ThreadSafeObjectProvider(Of MyApplication)
        Private Shared ReadOnly m_UserObjectProvider As ThreadSafeObjectProvider(Of User) = New ThreadSafeObjectProvider(Of User)
        Private Shared m_MyFormsObjectProvider As ThreadSafeObjectProvider(Of MyForms) = New ThreadSafeObjectProvider(Of MyForms)
        Private Shared ReadOnly m_MyWebServicesObjectProvider As ThreadSafeObjectProvider(Of MyWebServices) = New ThreadSafeObjectProvider(Of MyWebServices)

        ' Nested Types
        <EditorBrowsable(EditorBrowsableState.Never), MyGroupCollection("System.Windows.Forms.Form", "Create__Instance__", "Dispose__Instance__", "My.MyProject.Forms")> _
        Friend NotInheritable Class MyForms
            ' Methods
            <DebuggerHidden> _
            Private Shared Function Create__Instance__(Of T As { Form, New })(ByVal Instance As T) As T
                If ((Instance Is Nothing) OrElse Instance.IsDisposed) Then
                    If (Not MyForms.m_FormBeingCreated Is Nothing) Then
                        If MyForms.m_FormBeingCreated.ContainsKey(GetType(T)) Then
                            Throw New InvalidOperationException(Utils.GetResourceString("WinForms_RecursiveFormCreate", New String(0  - 1) {}))
                        End If
                    Else
                        MyForms.m_FormBeingCreated = New Hashtable
                    End If
                    MyForms.m_FormBeingCreated.Add(GetType(T), Nothing)
                    Try 
                        Return Activator.CreateInstance(Of T)
                    Catch obj1 As Object When (?)
                        Dim exception As TargetInvocationException
                        Throw New InvalidOperationException(Utils.GetResourceString("WinForms_SeeInnerException", New String() { exception.InnerException.Message }), exception.InnerException)
                    Finally
                        MyForms.m_FormBeingCreated.Remove(GetType(T))
                    End Try
                End If
                Return Instance
            End Function

            <DebuggerHidden> _
            Private Sub Dispose__Instance__(Of T As Form)(ByRef instance As T)
                instance.Dispose
                instance = CType(Nothing, T)
            End Sub

            <EditorBrowsable(EditorBrowsableState.Never)> _
            Public Overrides Function Equals(ByVal o As Object) As Boolean
                Return MyBase.Equals(RuntimeHelpers.GetObjectValue(o))
            End Function

            <EditorBrowsable(EditorBrowsableState.Never)> _
            Public Overrides Function GetHashCode() As Integer
                Return MyBase.GetHashCode
            End Function

            <EditorBrowsable(EditorBrowsableState.Never)> _
            Friend Function [GetType]() As Type
                Return GetType(MyForms)
            End Function

            <EditorBrowsable(EditorBrowsableState.Never)> _
            Public Overrides Function ToString() As String
                Return MyBase.ToString
            End Function


            ' Properties
            Public Property AboutBoxForm As AboutBoxForm
                Get
                    Me.m_AboutBoxForm = MyForms.Create__Instance__(Of AboutBoxForm)(Me.m_AboutBoxForm)
                    Return Me.m_AboutBoxForm
                End Get
                Set(ByVal Value As AboutBoxForm)
                    If (Not Value Is Me.m_AboutBoxForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of AboutBoxForm)(Me.m_AboutBoxForm)
                    End If
                End Set
            End Property

            Public Property CompileForm As CompileForm
                Get
                    Me.m_CompileForm = MyForms.Create__Instance__(Of CompileForm)(Me.m_CompileForm)
                    Return Me.m_CompileForm
                End Get
                Set(ByVal Value As CompileForm)
                    If (Not Value Is Me.m_CompileForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of CompileForm)(Me.m_CompileForm)
                    End If
                End Set
            End Property

            Public Property EscapeForm As EscapeForm
                Get
                    Me.m_EscapeForm = MyForms.Create__Instance__(Of EscapeForm)(Me.m_EscapeForm)
                    Return Me.m_EscapeForm
                End Get
                Set(ByVal Value As EscapeForm)
                    If (Not Value Is Me.m_EscapeForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of EscapeForm)(Me.m_EscapeForm)
                    End If
                End Set
            End Property

            Public Property GenerateCodeForm As GenerateCodeForm
                Get
                    Me.m_GenerateCodeForm = MyForms.Create__Instance__(Of GenerateCodeForm)(Me.m_GenerateCodeForm)
                    Return Me.m_GenerateCodeForm
                End Get
                Set(ByVal Value As GenerateCodeForm)
                    If (Not Value Is Me.m_GenerateCodeForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of GenerateCodeForm)(Me.m_GenerateCodeForm)
                    End If
                End Set
            End Property

            Public Property MainForm As MainForm
                Get
                    Me.m_MainForm = MyForms.Create__Instance__(Of MainForm)(Me.m_MainForm)
                    Return Me.m_MainForm
                End Get
                Set(ByVal Value As MainForm)
                    If (Not Value Is Me.m_MainForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of MainForm)(Me.m_MainForm)
                    End If
                End Set
            End Property

            Public Property PropertiesForm As PropertiesForm
                Get
                    Me.m_PropertiesForm = MyForms.Create__Instance__(Of PropertiesForm)(Me.m_PropertiesForm)
                    Return Me.m_PropertiesForm
                End Get
                Set(ByVal Value As PropertiesForm)
                    If (Not Value Is Me.m_PropertiesForm) Then
                        If (Not Value Is Nothing) Then
                            Throw New ArgumentException("Property can only be set to Nothing")
                        End If
                        Me.Dispose__Instance__(Of PropertiesForm)(Me.m_PropertiesForm)
                    End If
                End Set
            End Property


            ' Fields
            Public m_AboutBoxForm As AboutBoxForm
            Public m_CompileForm As CompileForm
            Public m_EscapeForm As EscapeForm
            Public m_GenerateCodeForm As GenerateCodeForm
            Public m_MainForm As MainForm
            Public m_PropertiesForm As PropertiesForm
            <ThreadStatic> _
            Private Shared m_FormBeingCreated As Hashtable
        End Class

        <EditorBrowsable(EditorBrowsableState.Never), MyGroupCollection("System.Web.Services.Protocols.SoapHttpClientProtocol", "Create__Instance__", "Dispose__Instance__", "")> _
        Friend NotInheritable Class MyWebServices
            ' Methods
            <DebuggerHidden> _
            Private Shared Function Create__Instance__(Of T As New)(ByVal instance As T) As T
                If (instance Is Nothing) Then
                    Return Activator.CreateInstance(Of T)
                End If
                Return instance
            End Function

            <DebuggerHidden> _
            Private Sub Dispose__Instance__(Of T)(ByRef instance As T)
                instance = CType(Nothing, T)
            End Sub

            <DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)> _
            Public Overrides Function Equals(ByVal o As Object) As Boolean
                Return MyBase.Equals(RuntimeHelpers.GetObjectValue(o))
            End Function

            <EditorBrowsable(EditorBrowsableState.Never), DebuggerHidden> _
            Public Overrides Function GetHashCode() As Integer
                Return MyBase.GetHashCode
            End Function

            <DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)> _
            Friend Function [GetType]() As Type
                Return GetType(MyWebServices)
            End Function

            <DebuggerHidden, EditorBrowsable(EditorBrowsableState.Never)> _
            Public Overrides Function ToString() As String
                Return MyBase.ToString
            End Function

        End Class

        <ComVisible(False), EditorBrowsable(EditorBrowsableState.Never)> _
        Friend NotInheritable Class ThreadSafeObjectProvider(Of T As New)
            ' Properties
            Friend ReadOnly Property GetInstance As T
                <DebuggerHidden> _
                Get
                    If (ThreadSafeObjectProvider(Of T).m_ThreadStaticValue Is Nothing) Then
                        ThreadSafeObjectProvider(Of T).m_ThreadStaticValue = Activator.CreateInstance(Of T)
                    End If
                    Return ThreadSafeObjectProvider(Of T).m_ThreadStaticValue
                End Get
            End Property


            ' Fields
            <ThreadStatic, CompilerGenerated> _
            Private Shared m_ThreadStaticValue As T
        End Class
    End Class
End Namespace

